cache-tree.c

```diff
@@ -757,27 +771,49 @@ static void prime_cache_tree_rec(struct repository *r,
 		else {
 			struct cache_tree_sub *sub;
 			struct tree *subtree = lookup_tree(r, &entry.oid);
+
 			if (!subtree->object.parsed)
 				parse_tree(subtree);
 			sub = cache_tree_sub(it, entry.path);
 			sub->cache_tree = cache_tree();
-			prime_cache_tree_rec(r, sub->cache_tree, subtree);
+			strbuf_reset(&subtree_path);
+			strbuf_grow(&subtree_path, tree_path->len + entry.pathlen + 1);
+			strbuf_addbuf(&subtree_path, tree_path);
+			strbuf_add(&subtree_path, entry.path, entry.pathlen);
+			strbuf_addch(&subtree_path, '/');
+
+			/*
+			* If a sparse index is in use, the directory being processed may be
+			* sparse. To confirm that, we can check whether an entry with that
+			* exact name exists in the index. If it does, the created subtree
+			* should be sparse. Otherwise, cache tree expansion should continue
+			* as normal.
+			*/
```

Comment:
nit: the spacing on this comment should have an extra space to have the `*` characters lined up vertically. VS Code's tab-in/out feature frequently loses this space.